dvc/render/convert.py

```diff
@@ -19,33 +24,26 @@ def _get_converter(
     raise ValueError(f"Invalid renderer class {renderer_class}")
 
 
-def _group_by_rev(datapoints):
-    grouped = defaultdict(list)
-    for datapoint in datapoints:
-        rev = datapoint.get(VERSION_FIELD, {}).get("revision")
-        grouped[rev].append(datapoint)
-    return dict(grouped)
-
-
 def to_json(renderer, split: bool = False) -> List[Dict]:
     if renderer.TYPE == "vega":
-        grouped = _group_by_rev(renderer.datapoints)
+        if not renderer.datapoints:
+            return []
         if split:
             content = renderer.get_filled_template(
                 skip_anchors=["data"], as_string=False
```

Comment:
[I] Need to skip `color`, `stroke_dash`, `shape` as we want to suppress the legend unless zoomed.

Reply:
is there a better way to do legend/zoom suppression?